crates/sui-framework/packages/sui-framework/sources/deny_list.move

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Defines the `DenyList` type. The `DenyList` shared object is used to restrict access to
+/// instances of certain core types from being used as inputs by specified addresses in the "deny
+/// list".
+module sui::deny_list {
+    use sui::tx_context::TxContext;
+    use sui::object::{Self, UID};
+    use sui::transfer;
+    use std::vector;
+    use sui::tx_context;
+    use sui::table::{Self, Table};
+    use sui::vec_set::{Self, VecSet};
+    use sui::versioned::{Self, Versioned};
+
+    friend sui::coin;
+
+    /// Trying to create a deny list object when not called by the system address.
+    const ENotSystemAddress: u64 = 0;
+    /// The specified address to be removed is not already in the deny list.
+    const ENotDenied: u64 = 1;
+
+    #[allow(unused_const)]
+    /// The index into the deny list vector for the `sui::coin::Coin` type.
+    const COIN_INDEX: u64 = 0;
```

Comment:
This is already defined in coin.move?

Reply:
We don't have public consts. And this one is unused.  I just figured we would eventually want them all in one place, though I can remove if you want.